rmc/utils/constraint.py

```diff
@@ -1310,45 +1318,350 @@ def create_context_with_oe(
     logger.info("Output OE-annotated dedup context HT fields: %s", set(ht.row))
 
 
-def reformat_annotations_for_release(ht: hl.Table) -> hl.Table:
+def annot_rmc_with_aa(ht: hl.Table, overwrite_temp: bool):
+    """
+    Annotate RMC results HT with amino acid information.
+
+    :param ht: Input RMC results HT.
+    :param overwrite_temp: Whether to overwrite temporary data.
+        If False, will read existing temp data rather than overwriting.
+        If True, will overwrite temp data.
+    :return: RMC results HT annotated with amino acid information.
+    """
+    logger.info("Getting amino acid information from context HT...")
+    rmc_transcripts = ht.aggregate(hl.agg.collect_as_set(ht.transcript))
+    context_ht = get_aa_from_context(
+        overwrite_temp=overwrite_temp, keep_transcripts=rmc_transcripts
+    )
+    context_ht = get_ref_aa(
+        ht=context_ht,
+        overwrite_temp=overwrite_temp,
+    )
+    ht = ht.annotate(
+        start_aa=context_ht[ht.start_coordinate, ht.transcript].ref_aa,
+        stop_aa=context_ht[ht.stop_coordinate, ht.transcript].ref_aa,
+    )
+    ht = ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/rmc_results_aa_annot.ht",
+        _read_if_exists=not overwrite_temp,
+        overwrite=overwrite_temp,
+    )
+    return check_and_fix_missing_aa(ht, context_ht, overwrite_temp)
+
+
+def join_and_fix_aa(ht: hl.Table, fix_ht: hl.Table) -> hl.Table:
+    """
+    Add start and stop amino acid information from fix_ht to ht.
+
+    :param ht: Input HT. Should be RMC results HT annotated with amino acid information.
+    :param fix_ht: HT start and stop amino acids only for RMC regions that were
+        previously missing amino acid information.
+    :return: RMC results HT annotated with amino acid information from fix HT.
+    """
+    ht = ht.annotate(
+        start_aa=hl.if_else(
+            hl.is_missing(ht.start_aa) & ht.transcript_start,
```

Comment:
Doesn't this only work when you call it after `fix_transcript_start_stops` and not `fix_region_start_stops`?

Reply:
good point, I'll add a note

Reply:
No I mean, this function as it is only adjusts start/stop AAs if they're at the start/end of transcripts but you use it in line 1584 to adjust region start/stop AAs too.

Reply:
oh shoot, yes